File: KthPrime.java
public class KthPrime {
    private static boolean isPrime(int n) {
        for (int i=2; i<n/2; i++) {
          if (n%i == 0)
            return false;
        }
        return true;
    }

    private static int kthPrime(int k) {
        int candidate = 2;
        while (k > 0) {
          if (isPrime(candidate))
            k--;
          candidate++;
        }
        return candidate-1;
    }

    public static void main(String[] args) {
        System.out.println(kthPrime(10000));
    }
}

File: kthprime.py
def is_prime(n):
  for num in range(2, n//2):
    if n%num == 0:
      return False
  return True

def kth_prime(k):
  candidate = 2
  while k:
    if is_prime(candidate):
      k -= 1
    candidate += 1
  return candidate - 1

# Driver code
print(kth_prime(10000)) # The 10000th prime number is 104723
